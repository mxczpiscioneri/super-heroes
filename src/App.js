import React from 'react'
import { StatusBar } from 'react-native'
import Routes from './routes'
import Colors from './statics/colors'
import FavoriteContext from './components/Favorite/context'

const App = () => (
  <FavoriteContext.Provider>
    <StatusBar barStyle="light-content" backgroundColor={Colors.SECONDARY} />
    <Routes />
  </FavoriteContext.Provider>
)

export default App
