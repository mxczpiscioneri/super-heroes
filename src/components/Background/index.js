import React from 'react'
import PropTypes from 'prop-types'
import { ImageBackground } from './styles'
import { BackgroundImage } from '../../statics/images'

export const Background = props => (
  <ImageBackground source={BackgroundImage}>
    {props.children}
  </ImageBackground>
)

Background.propTypes = {
  children: PropTypes.any,
}

Background.defaultprops = {
  children: '',
}
