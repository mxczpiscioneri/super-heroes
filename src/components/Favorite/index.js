import React from 'react'
import PropTypes from 'prop-types'
import {
  Icon,
  Image,
} from './styles'
import {
  IconFavorite,
  IconFavoriteBorder,
} from '../../statics/images'

export const Favorite = props => {
  const { character, favoritesIds, setFavorite } = props
  const isFavorite = favoritesIds.indexOf(character.id) >= 0

  return (
    <Icon activeOpacity={0.6} onPress={() => setFavorite(character)}>
      <Image source={isFavorite ? IconFavorite : IconFavoriteBorder} />
    </Icon>
  )
}

Favorite.propTypes = {
  character: PropTypes.object.isRequired,
  favoritesIds: PropTypes.array.isRequired,
  setFavorite: PropTypes.func.isRequired,
}
