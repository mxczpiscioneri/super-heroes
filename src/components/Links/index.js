import React from 'react'
import PropTypes from 'prop-types'
import { Linking } from 'react-native'
import {
  Container,
  Link,
  Text,
} from './styles'

export const Links = props => (
  <Container>
    {props.urls.map(item => (
      <Link
        key={item.type}
        activeOpacity={0.8}
        onPress={() => { Linking.openURL(item.url) }}
      >
        <Text>{item.type}</Text>
      </Link>
    ))}
  </Container>
)

Links.propTypes = {
  urls: PropTypes.array.isRequired,
}
