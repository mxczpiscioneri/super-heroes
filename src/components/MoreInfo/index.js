import React from 'react'
import PropTypes from 'prop-types'
import Carousel from 'react-native-snap-carousel'
import { Dimensions } from 'react-native'
import { generateUrlImage } from '../../utils'
import {
  Container,
  Title,
  Box,
  Image,
  Legend,
} from './styles'

const { width } = Dimensions.get('window')

export const MoreInfo = props => (
  <Container>
    <Title>{props.title}</Title>

    <Carousel
      data={props.data}
      sliderWidth={width}
      itemWidth={166}
      renderItem={({ item }) => (
        <Box key={item.id}>
          {item.thumbnail &&
            <Image source={{ uri: generateUrlImage(item.thumbnail.path, item.thumbnail.extension, 'portrait_xlarge') }} />
          }
          <Legend>{item.title}</Legend>
        </Box>
      )}
    />
  </Container>
)

MoreInfo.propTypes = {
  title: PropTypes.string.isRequired,
  data: PropTypes.array.isRequired,
}
