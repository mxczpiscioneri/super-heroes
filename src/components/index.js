export * from './Background'
export * from './Character'
export * from './Links'
export * from './MoreInfo'
