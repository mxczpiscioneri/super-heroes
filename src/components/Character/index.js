import React from 'react'
import PropTypes from 'prop-types'
import FavoriteContext from '../Favorite/context'
import {
  Container,
  Box,
  Image,
  Legend,
  FavoriteBox,
} from './styles'
import { Favorite } from '../Favorite'

export const Character = props => {
  const {
    character,
    image,
    onPress,
    legend,
  } = props

  return (
    <Container>
      <Box
        activeOpacity={0.9}
        onPress={onPress}
      >
        <Image source={{ uri: image }} />
        {legend &&
          <Legend>{character.name}</Legend>
        }
      </Box>
      <FavoriteBox>
        <FavoriteContext.Consumer>
          {context => (
            <Favorite {...context} character={character} />
          )}
        </FavoriteContext.Consumer>
      </FavoriteBox>
    </Container>
  )
}

Character.propTypes = {
  character: PropTypes.object.isRequired,
  image: PropTypes.string.isRequired,
  onPress: PropTypes.func,
  legend: PropTypes.bool,
}

Character.defaultprops = {
  onPress: null,
  legend: '',
}
