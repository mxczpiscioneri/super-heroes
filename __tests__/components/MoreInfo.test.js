
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { MoreInfo } from '../../src/components'
import { comicsMock } from '../../__mocks__/characters'

describe('MoreInfo', () => {
  it('MoreInfo snapshot', () => {
    const rendered = renderer.create(
      <MoreInfo title='Comics' data={comicsMock} />
    ).toJSON()
    expect(rendered).toMatchSnapshot()
  })

  it('MoreInfo renders correctly', () => {
    const title = 'Comics'
    const wrapper = shallow(<MoreInfo title={title} data={comicsMock} />)

    expect(wrapper.length).toEqual(1)
    expect(wrapper.find('Styled(Text)').length).toEqual(1)
    expect(wrapper.find('Styled(Text)').props().children).toEqual(title)
    expect(wrapper.find('Carousel').length).toEqual(1)
  })
})
