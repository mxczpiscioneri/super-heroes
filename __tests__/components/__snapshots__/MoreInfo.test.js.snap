// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoreInfo MoreInfo snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#1A3555",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "marginBottom": 16,
        "opacity": 0.9,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 24,
          "fontWeight": "bold",
          "marginVertical": 8,
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    Comics
  </Text>
  <RCTScrollView
    activeAnimationOptions={null}
    activeAnimationType="timing"
    activeSlideAlignment="center"
    activeSlideOffset={20}
    apparitionDelay={0}
    automaticallyAdjustContentInsets={false}
    autoplay={false}
    autoplayDelay={1000}
    autoplayInterval={3000}
    callbackOffsetMargin={5}
    containerCustomStyle={Object {}}
    contentContainerCustomStyle={Object {}}
    contentContainerStyle={
      Array [
        Object {
          "paddingLeft": 292,
          "paddingRight": 292,
        },
        Object {},
      ]
    }
    data={
      Array [
        Object {
          "characters": Object {
            "available": 8,
            "collectionURI": "http://gateway.marvel.com/v1/public/comics/22506/characters",
            "items": Array [
              Object {
                "name": "3-D Man",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1011334",
              },
              Object {
                "name": "Ant-Man (Eric O'Grady)",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1010802",
              },
              Object {
                "name": "Avengers",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1009165",
              },
              Object {
                "name": "Dum Dum Dugan",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1009284",
              },
              Object {
                "name": "Hank Pym",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1011490",
              },
              Object {
                "name": "Hardball",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1010821",
              },
              Object {
                "name": "Jocasta",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1009376",
              },
              Object {
                "name": "Komodo (Melati Kusuma)",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1010818",
              },
            ],
            "returned": 8,
          },
          "collectedIssues": Array [],
          "collections": Array [],
          "creators": Object {
            "available": 9,
            "collectionURI": "http://gateway.marvel.com/v1/public/comics/22506/creators",
            "items": Array [
              Object {
                "name": "Tom Brevoort",
                "resourceURI": "http://gateway.marvel.com/v1/public/creators/2133",
                "role": "editor",
              },
              Object {
                "name": "Virtual Calligr",
                "resourceURI": "http://gateway.marvel.com/v1/public/creators/452",
                "role": "letterer",
              },
              Object {
                "name": "Vc Chris Eliopoulos",
                "resourceURI": "http://gateway.marvel.com/v1/public/creators/12581",
                "role": "letterer",
              },
              Object {
                "name": "Bong Dazo",
                "resourceURI": "http://gateway.marvel.com/v1/public/creators/1400",
                "role": "penciller",
              },
              Object {
                "name": "Christos Gage",
                "resourceURI": "http://gateway.marvel.com/v1/public/creators/11765",
                "role": "writer",
              },
              Object {
                "name": "Dan Slott",
                "resourceURI": "http://gateway.marvel.com/v1/public/creators/12983",
                "role": "writer",
              },
              Object {
                "name": "Jay David Ramos",
                "resourceURI": "http://gateway.marvel.com/v1/public/creators/8706",
                "role": "colorist",
              },
              Object {
                "name": "Christopher Sotomayor",
                "resourceURI": "http://gateway.marvel.com/v1/public/creators/500",
                "role": "colorist",
              },
              Object {
                "name": "Harvey Tolibao",
                "resourceURI": "http://gateway.marvel.com/v1/public/creators/8584",
                "role": "inker",
              },
            ],
            "returned": 9,
          },
          "dates": Array [
            Object {
              "date": "2008-12-17T00:00:00-0500",
              "type": "onsaleDate",
            },
            Object {
              "date": "2008-11-27T00:00:00-0500",
              "type": "focDate",
            },
            Object {
              "date": "2010-02-23T00:00:00-0500",
              "type": "unlimitedDate",
            },
            Object {
              "date": "2011-08-09T00:00:00-0400",
              "type": "digitalPurchaseDate",
            },
          ],
          "description": "Join 3-D MAN, CLOUD 9, KOMODO, HARDBALL, and heroes around America in the battle that will decide the fate of the planet and the future of the Initiative program. Will the Kill Krew Army win the day?",
          "diamondCode": "SEP082362",
          "digitalId": 10949,
          "ean": "",
          "events": Object {
            "available": 1,
            "collectionURI": "http://gateway.marvel.com/v1/public/comics/22506/events",
            "items": Array [
              Object {
                "name": "Secret Invasion",
                "resourceURI": "http://gateway.marvel.com/v1/public/events/269",
              },
            ],
            "returned": 1,
          },
          "format": "Comic",
          "id": 22506,
          "images": Array [
            Object {
              "extension": "jpg",
              "path": "http://i.annihil.us/u/prod/marvel/i/mg/d/03/58dd080719806",
            },
          ],
          "isbn": "",
          "issn": "",
          "issueNumber": 19,
          "modified": "2015-10-27T16:38:23-0400",
          "pageCount": 32,
          "prices": Array [
            Object {
              "price": 2.99,
              "type": "printPrice",
            },
            Object {
              "price": 1.99,
              "type": "digitalPurchasePrice",
            },
          ],
          "resourceURI": "http://gateway.marvel.com/v1/public/comics/22506",
          "series": Object {
            "name": "Avengers: The Initiative (2007 - 2010)",
            "resourceURI": "http://gateway.marvel.com/v1/public/series/1945",
          },
          "stories": Object {
            "available": 2,
            "collectionURI": "http://gateway.marvel.com/v1/public/comics/22506/stories",
            "items": Array [
              Object {
                "name": "AVENGERS: THE INITIATIVE (2007) #19",
                "resourceURI": "http://gateway.marvel.com/v1/public/stories/49888",
                "type": "cover",
              },
              Object {
                "name": "Avengers: The Initiative (2007) #19 - Int",
                "resourceURI": "http://gateway.marvel.com/v1/public/stories/49889",
                "type": "interiorStory",
              },
            ],
            "returned": 2,
          },
          "textObjects": Array [
            Object {
              "language": "en-us",
              "text": "Join 3-D MAN, CLOUD 9, KOMODO, HARDBALL, and heroes around America in the battle that will decide the fate of the planet and the future of the Initiative program. Will the Kill Krew Army win the day?",
              "type": "issue_preview_text",
            },
            Object {
              "language": "en-us",
              "text": "SECRET INVASION Tie-In!
<br>'V-S DAY'
<br>It's been leading to this since the Hank Pym Skrull first came up with the idea for a Fifty State Initiative.  This is the final assault in the Secret Invasion, a nation-wide plan that will test the limits of 3-D MAN'S superhuman militia, THE KILL KREW ARMY! Join 3-D MAN, CLOUD 9, KOMODO, HARDBALL, and heroes around America in the battle that will decide the fate of the planet and the future of the Initiative program.  Win or lose, there's no turning back.  After today, everything changes.
<br>Rated T+ ...$2.99
<br>",
              "type": "issue_solicit_text",
            },
          ],
          "thumbnail": Object {
            "extension": "jpg",
            "path": "http://i.annihil.us/u/prod/marvel/i/mg/d/03/58dd080719806",
          },
          "title": "Avengers: The Initiative (2007) #19",
          "upc": "5960606084-01911",
          "urls": Array [
            Object {
              "type": "detail",
              "url": "http://marvel.com/comics/issue/22506/avengers_the_initiative_2007_19?utm_campaign=apiRef&utm_source=5da970115b50a3e9868cb1299a8e2bab",
            },
            Object {
              "type": "purchase",
              "url": "http://comicstore.marvel.com/Avengers-The-Initiative-19/digital-comic/10949?utm_campaign=apiRef&utm_source=5da970115b50a3e9868cb1299a8e2bab",
            },
            Object {
              "type": "reader",
              "url": "http://marvel.com/digitalcomics/view.htm?iid=10949&utm_campaign=apiRef&utm_source=5da970115b50a3e9868cb1299a8e2bab",
            },
            Object {
              "type": "inAppLink",
              "url": "https://applink.marvel.com/issue/10949?utm_campaign=apiRef&utm_source=5da970115b50a3e9868cb1299a8e2bab",
            },
          ],
          "variantDescription": "",
          "variants": Array [],
        },
        Object {
          "characters": Object {
            "available": 12,
            "collectionURI": "http://gateway.marvel.com/v1/public/comics/22299/characters",
            "items": Array [
              Object {
                "name": "3-D Man",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1011334",
              },
              Object {
                "name": "Ant-Man (Eric O'Grady)",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1010802",
              },
              Object {
                "name": "Avengers",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1009165",
              },
              Object {
                "name": "Cloud 9",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1010823",
              },
              Object {
                "name": "Dum Dum Dugan",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1009284",
              },
              Object {
                "name": "Gravity",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1010702",
              },
              Object {
                "name": "Hank Pym",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1011490",
              },
              Object {
                "name": "Hardball",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1010821",
              },
              Object {
                "name": "Jocasta",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1009376",
              },
              Object {
                "name": "Komodo (Melati Kusuma)",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1010818",
              },
              Object {
                "name": "Skrulls",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1009599",
              },
              Object {
                "name": "Thor Girl",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1010820",
              },
            ],
            "returned": 12,
          },
          "collectedIssues": Array [],
          "collections": Array [],
          "creators": Object {
            "available": 7,
            "collectionURI": "http://gateway.marvel.com/v1/public/comics/22299/creators",
            "items": Array [
              Object {
                "name": "Tom Brevoort",
                "resourceURI": "http://gateway.marvel.com/v1/public/creators/2133",
                "role": "editor",
              },
              Object {
                "name": "Vc Joe Caramagna",
                "resourceURI": "http://gateway.marvel.com/v1/public/creators/5251",
                "role": "letterer",
              },
              Object {
                "name": "Christos Gage",
                "resourceURI": "http://gateway.marvel.com/v1/public/creators/11765",
                "role": "writer",
              },
              Object {
                "name": "Dan Slott",
                "resourceURI": "http://gateway.marvel.com/v1/public/creators/12983",
                "role": "writer",
              },
              Object {
                "name": "Andrew Hennessy",
                "resourceURI": "http://gateway.marvel.com/v1/public/creators/548",
                "role": "inker",
              },
              Object {
                "name": "Steve Kurth",
                "resourceURI": "http://gateway.marvel.com/v1/public/creators/4981",
                "role": "penciller",
              },
              Object {
                "name": "Matt Milla",
                "resourceURI": "http://gateway.marvel.com/v1/public/creators/1405",
                "role": "colorist",
              },
            ],
            "returned": 7,
          },
          "dates": Array [
            Object {
              "date": "2008-10-29T00:00:00-0400",
              "type": "onsaleDate",
            },
            Object {
              "date": "2008-10-09T00:00:00-0400",
              "type": "focDate",
            },
            Object {
              "date": "2010-02-09T00:00:00-0500",
              "type": "unlimitedDate",
            },
            Object {
              "date": "2011-08-09T00:00:00-0400",
              "type": "digitalPurchaseDate",
            },
          ],
          "description": "Now that the Kill Krew knows Skrullowjacket's master plan, can they stop the true mission of the Fifty State Initiative? Plus, Thor Girl vs. Ultra Girl! One is more than she appears to be and the other's a Skrull!",
          "diamondCode": "AUG082377",
          "digitalId": 10948,
          "ean": "",
          "events": Object {
            "available": 1,
            "collectionURI": "http://gateway.marvel.com/v1/public/comics/22299/events",
            "items": Array [
              Object {
                "name": "Secret Invasion",
                "resourceURI": "http://gateway.marvel.com/v1/public/events/269",
              },
            ],
            "returned": 1,
          },
          "format": "Comic",
          "id": 22299,
          "images": Array [
            Object {
              "extension": "jpg",
              "path": "http://i.annihil.us/u/prod/marvel/i/mg/6/20/58dd057d304d1",
            },
            Object {
              "extension": "jpg",
              "path": "http://i.annihil.us/u/prod/marvel/i/mg/9/20/4bb6d098d144a",
            },
          ],
          "isbn": "",
          "issn": "",
          "issueNumber": 18,
          "modified": "2014-08-05T14:09:33-0400",
          "pageCount": 32,
          "prices": Array [
            Object {
              "price": 2.99,
              "type": "printPrice",
            },
            Object {
              "price": 1.99,
              "type": "digitalPurchasePrice",
            },
          ],
          "resourceURI": "http://gateway.marvel.com/v1/public/comics/22299",
          "series": Object {
            "name": "Avengers: The Initiative (2007 - 2010)",
            "resourceURI": "http://gateway.marvel.com/v1/public/series/1945",
          },
          "stories": Object {
            "available": 2,
            "collectionURI": "http://gateway.marvel.com/v1/public/comics/22299/stories",
            "items": Array [
              Object {
                "name": "AVENGERS: THE INITIATIVE (2007) #18",
                "resourceURI": "http://gateway.marvel.com/v1/public/stories/49103",
                "type": "cover",
              },
              Object {
                "name": "Avengers: The Initiative (2007) #18 - Int",
                "resourceURI": "http://gateway.marvel.com/v1/public/stories/49104",
                "type": "interiorStory",
              },
            ],
            "returned": 2,
          },
          "textObjects": Array [
            Object {
              "language": "en-us",
              "text": "Now that the Kill Krew knows Skrullowjacket's master plan, can they stop the true mission of the Fifty State Initiative? Plus, Thor Girl vs. Ultra Girl! One is more than she appears to be and the other's a Skrull!",
              "type": "issue_preview_text",
            },
            Object {
              "language": "en-us",
              "text": "SECRET INVASION TIE-IN!
<br>THE EXPLOSIVE FINALE STARTS HERE!
<br>Now that the KILL KREW knows Skrullowjacket's master plan, can they stop the TRUE purpose of the Fifty State Initiative? Plus:  It's THOR GIRL vs. ULTRA GIRL!  One is more than she appears to be... and the other's a Skrull. And, after fourteen issues, are we REALLY going to unmask MUTANT ZERO?!
<br>Rated T+ ...$2.99",
              "type": "issue_solicit_text",
            },
          ],
          "thumbnail": Object {
            "extension": "jpg",
            "path": "http://i.annihil.us/u/prod/marvel/i/mg/6/20/58dd057d304d1",
          },
          "title": "Avengers: The Initiative (2007) #18",
          "upc": "5960606084-01811",
          "urls": Array [
            Object {
              "type": "detail",
              "url": "http://marvel.com/comics/issue/22299/avengers_the_initiative_2007_18?utm_campaign=apiRef&utm_source=5da970115b50a3e9868cb1299a8e2bab",
            },
            Object {
              "type": "purchase",
              "url": "http://comicstore.marvel.com/Avengers-The-Initiative-18/digital-comic/10948?utm_campaign=apiRef&utm_source=5da970115b50a3e9868cb1299a8e2bab",
            },
            Object {
              "type": "reader",
              "url": "http://marvel.com/digitalcomics/view.htm?iid=10948&utm_campaign=apiRef&utm_source=5da970115b50a3e9868cb1299a8e2bab",
            },
            Object {
              "type": "inAppLink",
              "url": "https://applink.marvel.com/issue/10948?utm_campaign=apiRef&utm_source=5da970115b50a3e9868cb1299a8e2bab",
            },
          ],
          "variantDescription": "",
          "variants": Array [
            Object {
              "name": "Avengers: The Initiative (2007) #18 (ZOMBIE VARIANT)",
              "resourceURI": "http://gateway.marvel.com/v1/public/comics/22300",
            },
          ],
        },
        Object {
          "characters": Object {
            "available": 9,
            "collectionURI": "http://gateway.marvel.com/v1/public/comics/22300/characters",
            "items": Array [
              Object {
                "name": "3-D Man",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1011334",
              },
              Object {
                "name": "Ant-Man (Eric O'Grady)",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1010802",
              },
              Object {
                "name": "Cloud 9",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1010823",
              },
              Object {
                "name": "Dum Dum Dugan",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1009284",
              },
              Object {
                "name": "Gravity",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1010702",
              },
              Object {
                "name": "Hank Pym",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1011490",
              },
              Object {
                "name": "Komodo (Melati Kusuma)",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1010818",
              },
              Object {
                "name": "Skrulls",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1009599",
              },
              Object {
                "name": "Slapstick",
                "resourceURI": "http://gateway.marvel.com/v1/public/characters/1011057",
              },
            ],
            "returned": 9,
          },
          "collectedIssues": Array [],
          "collections": Array [],
          "creators": Object {
            "available": 8,
            "collectionURI": "http://gateway.marvel.com/v1/public/comics/22300/creators",
            "items": Array [
              Object {
                "name": "Tom Brevoort",
                "resourceURI": "http://gateway.marvel.com/v1/public/creators/2133",
                "role": "editor",
              },
              Object {
                "name": "Vc Joe Caramagna",
                "resourceURI": "http://gateway.marvel.com/v1/public/creators/5251",
                "role": "letterer",
              },
              Object {
                "name": "Juan Doe",
                "resourceURI": "http://gateway.marvel.com/v1/public/creators/1054",
                "role": "penciller (cover)",
              },
              Object {
                "name": "Christos Gage",
                "resourceURI": "http://gateway.marvel.com/v1/public/creators/11765",
                "role": "writer",
              },
              Object {
                "name": "Dan Slott",
                "resourceURI": "http://gateway.marvel.com/v1/public/creators/12983",
                "role": "writer",
              },
              Object {
                "name": "Andrew Hennessy",
                "resourceURI": "http://gateway.marvel.com/v1/public/creators/548",
                "role": "inker",
              },
              Object {
                "name": "Steve Kurth",
                "resourceURI": "http://gateway.marvel.com/v1/public/creators/4981",
                "role": "penciller",
              },
              Object {
                "name": "Matt Milla",
                "resourceURI": "http://gateway.marvel.com/v1/public/creators/1405",
                "role": "colorist",
              },
            ],
            "returned": 8,
          },
          "dates": Array [
            Object {
              "date": "2008-10-29T00:00:00-0400",
              "type": "onsaleDate",
            },
            Object {
              "date": "2008-10-09T00:00:00-0400",
              "type": "focDate",
            },
          ],
          "description": "SECRET INVASION TIE-IN!
<br>THE EXPLOSIVE FINALE STARTS HERE!
<br>Now that the KILL KREW knows Skrullowjacket's master plan, can they stop the TRUE purpose of the Fifty State Initiative? Plus:  It's THOR GIRL vs. ULTRA GIRL!  One is more than she appears to be... and the other's a Skrull. And, after fourteen issues, are we REALLY going to unmask MUTANT ZERO?!
<br>Rated T+ ...$2.99
<br>",
          "diamondCode": "AUG082378",
          "digitalId": 0,
          "ean": "",
          "events": Object {
            "available": 0,
            "collectionURI": "http://gateway.marvel.com/v1/public/comics/22300/events",
            "items": Array [],
            "returned": 0,
          },
          "format": "Comic",
          "id": 22300,
          "images": Array [
            Object {
              "extension": "jpg",
              "path": "http://i.annihil.us/u/prod/marvel/i/mg/1/10/4e94a23255996",
            },
            Object {
              "extension": "jpg",
              "path": "http://i.annihil.us/u/prod/marvel/i/mg/b/20/4bb63aa561aa0",
            },
          ],
          "isbn": "",
          "issn": "",
          "issueNumber": 18,
          "modified": "-0001-11-30T00:00:00-0500",
          "pageCount": 32,
          "prices": Array [
            Object {
              "price": 2.99,
              "type": "printPrice",
            },
          ],
          "resourceURI": "http://gateway.marvel.com/v1/public/comics/22300",
          "series": Object {
            "name": "Avengers: The Initiative (2007 - 2010)",
            "resourceURI": "http://gateway.marvel.com/v1/public/series/1945",
          },
          "stories": Object {
            "available": 2,
            "collectionURI": "http://gateway.marvel.com/v1/public/comics/22300/stories",
            "items": Array [
              Object {
                "name": "Avengers: The Initiative (2007) #18, Zombie Variant",
                "resourceURI": "http://gateway.marvel.com/v1/public/stories/49105",
                "type": "cover",
              },
              Object {
                "name": "Avengers: The Initiative (2007) #18, Zombie Variant - Int",
                "resourceURI": "http://gateway.marvel.com/v1/public/stories/49106",
                "type": "interiorStory",
              },
            ],
            "returned": 2,
          },
          "textObjects": Array [
            Object {
              "language": "en-us",
              "text": "SECRET INVASION TIE-IN!
<br>THE EXPLOSIVE FINALE STARTS HERE!
<br>Now that the KILL KREW knows Skrullowjacket's master plan, can they stop the TRUE purpose of the Fifty State Initiative? Plus:  It's THOR GIRL vs. ULTRA GIRL!  One is more than she appears to be... and the other's a Skrull. And, after fourteen issues, are we REALLY going to unmask MUTANT ZERO?!
<br>Rated T+ ...$2.99
<br>",
              "type": "issue_solicit_text",
            },
          ],
          "thumbnail": Object {
            "extension": "jpg",
            "path": "http://i.annihil.us/u/prod/marvel/i/mg/1/10/4e94a23255996",
          },
          "title": "Avengers: The Initiative (2007) #18 (ZOMBIE VARIANT)",
          "upc": "5960606084-01821",
          "urls": Array [
            Object {
              "type": "detail",
              "url": "http://marvel.com/comics/issue/22300/avengers_the_initiative_2007_18_zombie_variant/zombie_variant?utm_campaign=apiRef&utm_source=5da970115b50a3e9868cb1299a8e2bab",
            },
          ],
          "variantDescription": "ZOMBIE VARIANT",
          "variants": Array [
            Object {
              "name": "Avengers: The Initiative (2007) #18",
              "resourceURI": "http://gateway.marvel.com/v1/public/comics/22299",
            },
          ],
        },
      ]
    }
    decelerationRate="fast"
    directionalLockEnabled={true}
    disableVirtualization={false}
    enableMomentum={false}
    enableSnap={true}
    firstItem={0}
    getItem={[Function]}
    getItemCount={[Function]}
    hasParallaxImages={false}
    horizontal={true}
    inactiveSlideOpacity={0.7}
    inactiveSlideScale={0.9}
    inactiveSlideShift={0}
    initialNumToRender={10}
    invertStickyHeaders={false}
    inverted={false}
    itemWidth={166}
    keyExtractor={[Function]}
    layout="default"
    lockScrollTimeoutDuration={1000}
    lockScrollWhileSnapping={false}
    loop={false}
    loopClonesPerSide={3}
    maxToRenderPerBatch={21}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onResponderRelease={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    overScrollMode="never"
    pinchGestureEnabled={false}
    removeClippedSubviews={true}
    renderItem={[Function]}
    scrollEnabled={true}
    scrollEventThrottle={1}
    scrollsToTop={false}
    shouldOptimizeUpdates={true}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    slideStyle={Object {}}
    sliderWidth={750}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flexDirection": "row",
        "opacity": 0,
        "width": 750,
      }
    }
    swipeThreshold={20}
    updateCellsBatchingPeriod={50}
    useScrollView={false}
    vertical={false}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "opacity": 1,
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
              "width": 166,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 8,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 8,
                },
              ]
            }
          >
            <Image
              source={
                Object {
                  "uri": "http://i.annihil.us/u/prod/marvel/i/mg/d/03/58dd080719806/portrait_xlarge.jpg",
                }
              }
              style={
                Array [
                  Object {
                    "borderRadius": 6,
                    "height": 225,
                    "width": 150,
                  },
                ]
              }
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 16,
                    "marginTop": 8,
                  },
                ]
              }
            >
              Avengers: The Initiative (2007) #19
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "opacity": 0.7,
              "transform": Array [
                Object {
                  "scale": 0.9,
                },
              ],
              "width": 166,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 8,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 8,
                },
              ]
            }
          >
            <Image
              source={
                Object {
                  "uri": "http://i.annihil.us/u/prod/marvel/i/mg/6/20/58dd057d304d1/portrait_xlarge.jpg",
                }
              }
              style={
                Array [
                  Object {
                    "borderRadius": 6,
                    "height": 225,
                    "width": 150,
                  },
                ]
              }
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 16,
                    "marginTop": 8,
                  },
                ]
              }
            >
              Avengers: The Initiative (2007) #18
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "opacity": 0.7,
              "transform": Array [
                Object {
                  "scale": 0.9,
                },
              ],
              "width": 166,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 8,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 8,
                },
              ]
            }
          >
            <Image
              source={
                Object {
                  "uri": "http://i.annihil.us/u/prod/marvel/i/mg/1/10/4e94a23255996/portrait_xlarge.jpg",
                }
              }
              style={
                Array [
                  Object {
                    "borderRadius": 6,
                    "height": 225,
                    "width": 150,
                  },
                ]
              }
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 16,
                    "marginTop": 8,
                  },
                ]
              }
            >
              Avengers: The Initiative (2007) #18 (ZOMBIE VARIANT)
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
